src/main/java/be/uantwerpen/minelabs/potion/GasPotion.java

```diff
@@ -27,8 +30,11 @@ public GasPotion(Settings settings, Molecule molecule) {
     @Override
     public void appendStacks(ItemGroup group, DefaultedList<ItemStack> stacks) {
         Potion potion = new Potion("Test");
```

Comment:
Is this used anywhere? Otherwise I think you can skip this entire thing rather than first setting nbt with potion util and then setting it to null.